lib/ansible/modules/system/user.py

```diff
@@ -514,7 +526,13 @@ def modify_user_usermod(self):
 
         if self.expires:
             cmd.append('--expiredate')
-            cmd.append(time.strftime(self.DATE_FORMAT, self.expires))
+            try:
+                if self.clearexpires < 0:
+                    cmd.append('')
```

Comment:
Yeah, but this only works because `cmd` is a list and appending `''` to that list will insert an empty string as a param. 

The second thing is the missing idempotence of expires (always changed): It results from the users module not being capable of retrieving the current setting of expiredate, so you can compare it with the new setting and only set it, if it is changed.

With the second issue, the third quirk results: When you get the current setting of expiredate using `chage -l` for instance, it will give you a date-time-string according to some local setting, which is referring to a usermod setting.
So you need to make sure you use a common date format for all of this: Maybe thats why the current implementation just uses timestamps.